### ISSUE TITLE
Improper deduping when for values have different casing, and anchor is from anchor block

### ISSUE THREAD

**Issue by tabatkins** [speced/bikeshed#2837]
Example document:

```bikeshed
<pre class='metadata'>
Title: test
Shortname: test
Status: DREAM
Editor: foo
Abstract: foo
Die On: nothing
</pre>

<pre class="anchors">
spec: dom; urlPrefix: https://dom.spec.whatwg.org/
    type: dfn
        text: origin; for: document; url: concept-document-origin
</pre>

[=Document/origin=].

[=document/origin=].
```

Giving a "duplicated ID" warning, because it generates two identical entries in the index.

**Comment by tabatkins** [comment]
Ah, this is because DOM actually defines this reference, with for=Document, but the anchor-block defines it with for=document. So the two autolinks resolve to different references (so no deduping), but the two references are identical (so the generated ID ends up being identical).

**Comment by jyasskin** [comment]
i.e., this should probably be an error, just a more comprehensible one. :)

**Comment by tabatkins** [comment]
Indeed, and this is actually nice because I've been wondering how to help authors identify anchor-block refs they can remove. This problem suggests that I can just use similar code to what I have for generating IDs to identify "unnecessary" anchors. Not trivial, but easier than what I was imagining I'd have to do.

**Comment by tabatkins** [comment]
Closely related issue: if you add an anchor-block entry for a ref that already exists, but use a different spec name for it, you'll end up with two entries in your Index that also have identical generated IDs.
